runtimes/eden/src/migrations.rs

```diff
@@ -0,0 +1,103 @@
+use frame_support::traits::{OnRuntimeUpgrade, StorageVersion};
+
+use frame_support::weights::Weight;
+use sp_core::Get;
+#[cfg(feature = "try-runtime")]
+use {sp_runtime::TryRuntimeError, sp_std::prelude::*};
+
+pub struct MultiMigration<T>(sp_std::marker::PhantomData<T>);
+
+impl<T> OnRuntimeUpgrade for MultiMigration<T>
+where
+	T: pallet_scheduler::Config
+		+ pallet_balances::Config
+		+ pallet_membership::Config<pallet_membership::pallet::Instance3>
+		+ pallet_collective::Config<pallet_collective::pallet::Instance1>
+		+ pallet_collator_selection::Config
+		+ pallet_xcm::Config
+		+ pallet_preimage::Config
+		+ pallet_multisig::Config
+		+ pallet_contracts::Config,
+{
+	fn on_runtime_upgrade() -> Weight {
```

Comment:
@aliXsed please check if the reasoning below holds.

Reply:
@simonsso for those you didn't find any keys and you have only changed the storage version, you still need to run the migration logic if they have any. The reason being is we cannot guarantee there wouldn't be any keys inserted for those pallets before we run the actual upgrade.

Reply:
Unless the version deployed on mainnet is already the very latest for that pallet

Reply:
Checked all the pallets in substrate. XCM had been checked manually before.